PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx

```diff
@@ -119,168 +119,175 @@ struct HfCorrelatorDplusHadrons {
           continue;
         }
-        if (std::abs(track.dcaXY()) >= dcaXYTrackMax || std::abs(track.dcaZ()) >= dcaZTrackMax) {
-          continue; // Remove secondary tracks
+        if (candidate1.pt() > ptTrackMax) {
+          continue;
         }
-        // Removing Dplus daughters by checking track indices
-        if ((candidate1.prong0Id() == track.mRowIndex) || (candidate1.prong1Id() == track.mRowIndex) || (candidate1.prong2Id() == track.mRowIndex)) {
+        // check decay channel flag for candidate1
+        if (!(candidate1.hfflag() & 1 << DecayType::DplusToPiKPi)) {
           continue;
         }
-        entryDplusHadronPair(getDeltaPhi(track.phi(), candidate1.phi()),
-                             track.eta() - candidate1.eta(),
-                             candidate1.pt(),
-                             track.pt());
-        entryDplusHadronRecoInfo(invMassDplusToPiKPi(candidate1), 0);
-      } // Hadron Tracks loop
-    }   // end outer Dplus loop
+        double efficiencyWeight = 1.;
+        if (applyEfficiency) {
+          efficiencyWeight = 1. / efficiencyD->at(o2::analysis::findBin(binsPt, candidate1.pt()));
+        }
+        // fill invariant mass plots and generic info from all Dplus candidates
+        registry.fill(HIST("hMassDplus_2D"), invMassDplusToPiKPi(candidate1), candidate1.pt(), efficiencyWeight);
+        registry.fill(HIST("hMassDplusData"), invMassDplusToPiKPi(candidate1), efficiencyWeight);
+        registry.fill(HIST("hPtCand"), candidate1.pt());
+        registry.fill(HIST("hPtProng0"), candidate1.ptProng0());
+        registry.fill(HIST("hPtProng1"), candidate1.ptProng1());
+        registry.fill(HIST("hPtProng2"), candidate1.ptProng2());
+        registry.fill(HIST("hEta"), candidate1.eta());
+        registry.fill(HIST("hPhi"), RecoDecay::constrainAngle(candidate1.phi(), -o2::constants::math::PI / 2.));
+        registry.fill(HIST("hY"), yDplus(candidate1));
+        registry.fill(HIST("hSelectionStatus"), candidate1.isSelDplusToPiKPi());
+
+        // Dplus-Hadron correlation dedicated section
+        // if the candidate is a Dplus, search for Hadrons and evaluate correlations
+        for (const auto& track : tracks) {
+          if (std::abs(track.eta()) > etaTrackMax) {
+            continue;
+          }
+          if (track.pt() < ptTrackMin) {
+            continue;
+          }
+          if (std::abs(track.dcaXY()) >= dcaXYTrackMax || std::abs(track.dcaZ()) >= dcaZTrackMax) {
+            continue; // Remove secondary tracks
+          }
+          // Removing Dplus daughters by checking track indices
+          if ((candidate1.prong0Id() == track.globalIndex()) || (candidate1.prong1Id() == track.globalIndex()) || (candidate1.prong2Id() == track.globalIndex())) {
+            continue;
+          }
+          entryDplusHadronPair(RecoDecay::constrainAngle(getDeltaPhi(track.phi(), candidate1.phi()), -o2::constants::math::PI / 2.),
```

Comment:
Here you can remove the RecoDecay::constrainAngle call, since it's already contained within the getDeltaPhi method

Reply:
Thanks for the comment, I fixed the comment.